src/python/pants/backend/python/target_types.py

```diff
@@ -52,7 +52,8 @@
 
 
 class PythonSources(Sources):
-    expected_file_extensions = (".py", ".pyi")
+    # Note that Python scripts often have no file ending.
+    expected_file_extensions = ("", ".py", ".pyi")
```

Comment:
To clarify, this is just to allow such files, but doesn't change the default globs for sources= in a `python_library`?

Reply:
Correct. This is validation that we apply after hydration, meant to catch mistakes like `sources=['**']` resulting in unrelated files being included in the target, e.g. Java files. Otherwise, Pants will happily think it's Python and try to run, say, Black on it. Better to have our carefully crafted error message.